streams/src/main/java/org/apache/kafka/streams/state/internals/LRUCacheEntry.java

```diff
@@ -42,15 +41,17 @@
                   final long timestamp,
                   final int partition,
                   final String topic) {
-        super(timestamp, offset, partition, topic, headers);
-        this.value = value;
+        final ProcessorRecordContext context = new ProcessorRecordContext(timestamp, offset, partition, topic, headers);
+
+        this.record = new ContextualRecord<>(
+            value,
+            context
+        );
+
         this.isDirty = isDirty;
-        this.sizeBytes = (value == null ? 0 : value.length) +
-                1 + // isDirty
-                8 + // timestamp
-                8 + // offset
-                4 + // partition
-                (topic == null ? 0 : topic.length());
+        this.sizeBytes = 1 + // isDirty
+            (value == null ? 0 : value.length) +
```

Comment:
Should we add 4 byte to store array size? Also, do we have object overhead for an array type?

Reply:
as above.